tests/regression/matmul/main.cpp

```diff
@@ -0,0 +1,348 @@
+  //1
+  std::cout << "Debug :: num_tasks = " << num_tasks << std::endl;
+  kernel_arg.num_tasks = num_tasks;
+  kernel_arg.num_warps = num_warps;
+  kernel_arg.num_threads = num_threads;
+  kernel_arg.TC_per_warp = TC_per_warp;
+  //1
+  kernel_arg.matrix_size = matrix_size;
+  kernel_arg.data_size = data_size;
+  kernel_arg.tc_size = tc_size;
+
+  std::cout << "dev_src0=0x" << std::hex << kernel_arg.src0_addr << std::endl;
+  std::cout << "dev_src1=0x" << std::hex << kernel_arg.src1_addr << std::endl;
+  std::cout << "dev_dst=0x" << std::hex << kernel_arg.dst_addr << std::endl;
+
+  //////////////////////////////////////////////////  
+  //////////////////////////////////////////////////
+  
+  // upload kernel argument
+  std::cout << "upload kernel argument" << std::endl;
+  RT_CHECK(vx_upload_bytes(device, &kernel_arg, sizeof(kernel_arg_t), &args_buffer));
+
+  auto time_start = std::chrono::high_resolution_clock::now();
+
+    std::cout << "start device" << std::endl;
+    RT_CHECK(vx_start(device, krnl_buffer, args_buffer));
+
+    // wait for completion
+    std::cout << "wait for completion" << std::endl;
+    RT_CHECK(vx_ready_wait(device, VX_MAX_TIMEOUT));
+
+  auto time_end = std::chrono::high_resolution_clock::now();
+  double elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(time_end - time_start).count();
+  printf("Elapsed time: %lg ms\n", elapsed);
+
+  // download destination buffer
+  std::cout << "download destination buffer" << std::endl;
+  RT_CHECK(vx_copy_from_dev((int8_t*)variables.B_mat.data(), C_buffer, 0, buf_size));
+
+  // verify result (TODO : needs to be fixed for for functional correctness)
+  /*
+  std::cout << "verify result" << std::endl;  
+  {
+    int errors = 0;
+    auto buf_ptr = (int8_t*)staging_buf.data();
+    uint64_t tc_size = kernel_arg.tc_size;
+    std::cout << "tc_size = " << tc_size << std::endl;
+    int Result[matrix_size*matrix_size];
+    int n_tiles = (matrix_size/tc_size);
+    int tc_size_f = tc_size*tc_size;
+
+    //converting buf ptr (tile by tile) to CPU style linear (row by row)
+    for(int k = 0; k < matrix_size/tc_size; k+= 1)
+    {
+      for(int j = 0; j < matrix_size; j+= tc_size)
+      {
+        for(int i =0; i < tc_size*tc_size; i++)
+        {
+          Result[ tc_size*matrix_size*k +j+ (i/tc_size)*matrix_size +i%(tc_size)]  = buf_ptr[matrix_size*tc_size*k+tc_size*j+i];
+        }
+      }    
+    }
+
+    for (uint32_t i = 0; i < matrix_size*matrix_size; ++i) {
+      //int ref = i + i; 
+      int cur = Result[i];
+      if (cur != refs[i]) {
+        ++errors;
+      }
+    }
+    if (errors != 0) {
+      std::cout << "Found " << std::dec << errors << " errors!" << std::endl;
+      std::cout << "FAILED!" << std::endl;
+      return 1;  
+    }
+    else
+    {
+      std::cout << "CONDITIONALLY PASSED!" << std::endl;
+    }
+  }
+  */
+
+  // cleanup
+  std::cout << "cleanup" << std::endl;  
+  cleanup();
+
+  std::cout << "PASSED!" << std::endl;
```

Comment:
I see that you are printing the performance numbers but are those used in the regression test for performance regression?